src/SFA.DAS.Reservations.Domain/Reservations/Reservation.cs

```diff
@@ -79,8 +79,13 @@ public Reservation(Guid id, long accountId, DateTime startDate, int expiryPeriod
 
         private IList<Rule> GetRules(Func<DateTime, Task<IList<Rule>>> getRules)
         {
-            var task = getRules(StartDate);
-            return task.Result;
+            if (StartDate.HasValue)
```

Comment:
I can't find any tests for this if scenario, e.g. if no value, if has value

Reply:
Added tests to cover this scenario